File: minwook/17주차/귤_고르기.java
import java.util.*;
class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        int num = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int n : tangerine){
            map.put(n, map.getOrDefault(n, 0)+1);
        }
        List<Integer> list = new ArrayList<>(map.keySet());
        list.sort((o1, o2) -> map.get(o2) - map.get(o1));
        
        while(k > 0){
            k -= map.get(list.get(num));
            answer++;
            num++;
        }
        return answer;
    }
}

File: jeonghyeon/17week/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    const count = {}; // 귤 크기 개수 저장
    // 각 종류의 개수
    tangerine.forEach(i => { count[i] = ++count[i] || 1});
    const sort = Object.values(count).sort((a,b) => b-a); // 개수만 추출해 내림차순 정렬
    for(let i = 0; i < sort.length; i++) {
        if(k <= 0) {
            break;
        }
        k -= sort[i];
        answer++;
    }
    
    return answer;
}